### PR TITLE
DM-36260: Deferred charge trap array contains NaNs, butler access fails

### PR DESCRIPTION
**Pull request by czwa**
Fix handling of NaN padding in serial trap coefficients.
This version properly truncates the padded values, and filters incoming NaN values from the trap coefficients on construction.

### CODE REVIEW

File: python/lsst/ip/isr/deferredCharge.py [hunk 1]
```diff
@@ -378,7 +383,25 @@ def fromTable(cls, tableList):
...
-            ampTrap['coeffs'] = np.array(coeffs[index])[~np.isnan(coeffs[index])].tolist()
+
+            # Unpad any trailing NaN values: find the continuous array
+            # of NaNs at the end of the coefficients, and remove them.
+            inCoeffs = coeffs[index]
+            breakIndex = 1
+            nanValues = np.where(np.isnan(inCoeffs))[0]
+            if nanValues is not None:
+                coeffLength = len(inCoeffs)
+                while (breakIndex < coeffLength):
```
Comment (leeskelvin) [inline python/lsst/ip/isr/deferredCharge.py:394]: Does this need parentheses (just `while (breakIndex < coeffLength):`)?

File: python/lsst/ip/isr/deferredCharge.py [hunk 2]
```diff
@@ -378,7 +383,25 @@ def fromTable(cls, tableList):
...
+                    if coeffLength - breakIndex in nanValues:
+                        breakIndex += 1
+                    else:
+                        break
+            breakIndex -= 1  # Remove the fixed offset.
+            if breakIndex != 0:
+                outCoeffs = inCoeffs[0:coeffLength - breakIndex]
```
Comment (leeskelvin) [inline python/lsst/ip/isr/deferredCharge.py:401]: You might need a space either side of the `:` here.

Comment (czwa) [reply python/lsst/ip/isr/deferredCharge.py:401]: Flake8 didn't like the leading space, so I only added the one.

Comment (timj) [reply python/lsst/ip/isr/deferredCharge.py]: black will add the space but this package doesn't use black (which requires a tweak to flake8 settings).

**Pull request merged**